adaptive_scheduler/scheduler.py

```diff
@@ -869,11 +869,11 @@ def apply_rr_result(self, scheduler_result, scheduler_input, apply_deadline):
 
             # Cancel just the time slots under a newly scheduled RR
             n_deleted = self.clear_resource_schedules(cancelation_date_list_by_resource, include_rr=True,
-                                                      include_normal=True)
+                                                      include_normal=True, preemption_enabled=True)
             # Cancel any remaining RRs not under a newly scheduled RR (needed in case weather knocks out a telescope
             # that previously had a RR scheduled on it)
             n_deleted += self.clear_resource_schedules(all_cancelation_date_list_by_resource, include_rr=True,
```

Comment:
Should this also be wrapped in a `try/except` to still get to the code after it?

Reply:
since preemption is enabled, it can never reach the code path that will trigger a 400 for overlapping in progress observations

Reply:
I guess I'm confused... `preemption_enabled=False` in this call to `clear_resource_schedules`

Reply:
oh sorry I was looking at the line above. Yeah preemption enabled is false... hmm I guess you are right. The point of that call is only to delete rapid response requests that ARENT overlapping with the newly scheduled rapid response. But I suppose we could get into the same race condition with a rapid response request so we could see the same exception here. I'll add the try/except.

Reply:
oh wait, nvm! We already cancel all telescopes together for the rapid response run, rather than separated out by telescope, so there isn't really a reason to put a try/except block there since if we fail at cancelling we fail for cancelling all of the telescopes and should just restart anyway. We could separate out the cancelling for the rapid response run like we do for the normal run but I think that's not necessary...

Reply:
Hmm yeah I agree that seems unlikely to be necessary since there really aren't that many rapid response requests coming into the network. Ok so my one last question/ point of confusion, and I could totally be misunderstanding this, is from reading the comment above the call that might raise the error:
```
 # Cancel any remaining RRs not under a newly scheduled RR (needed in case weather knocks out a telescope
 # that previously had a RR scheduled on it)
```
Since the cancel call is getting rid of current RRs that are not interfering with any newly scheduled RRs, that makes me think that we would want to still proceed with the code after it to save the new RRs even if the call fails, since they won't be interfering and the new schedule that we generated is still valid. Although maybe the comment is misleading me. But maybe the point is that if the cancel call fails, then we might not go into an RR scheduling loop next time around and that RR we wanted to cancel might just end up hanging around.